src/rosdiscover/Issue74.py

```diff
@@ -0,0 +1,102 @@
+import time
+import collections
+import sys
+
+sys.path.append("/interpreter")
+from summary import NodeSummary
+
+
+from loguru import logger
+
+import roswire
+
+f = open("Arch.yml", "w")
+
+# enable logging
+logger.enable('roswire')
+
+
+# first, we create a new ROSWire session
+rsw = roswire.ROSWire()
+
+def toString(l): 
+    s = ''
+    l1 = list(l)
+    if len(l1) == 0: 
+        return '[]'
+    for i in l1:
+        (fmt, topic) = i
+        s = s + "\n  - format: " + fmt + "\n    name: " + topic + "\n" 
+    return s
+
+# we then launch a container for a given robot.
+# once the end of the given context is reached, including the event where an
+# exception occurs, the container will be automatically destroyed.
+image = 'therobotcooperative/turtlebot3'
+sources = ['/opt/ros/kinetic/setup.bash', '/ros_ws/devel/setup.bash']
+environment = {'TURTLEBOT3_MODEL': 'burger'}
+with rsw.launch(image, sources, environment=environment) as system:
+
+    # we then launch ROS inside the container
+    # again, once the end of the context is reached, ROS is killed.
+    with system.roscore() as ros:
+
+        # let's bring up the application
+        ros.roslaunch('turtlebot3_house.launch',
+                      package='turtlebot3_gazebo',
+                      args={'gui': 'false'})
+
+        # we need to wait for the nodes to finish their initialisation
+        time.sleep(30)
```

Comment:
We should work out if there is a better way to do this. Sleeping for a fixed amount isn't very general. @ChrisTimperley is there a more principled way to do this generally in ROS? I seem to recall that it is complicated.